pkg/webhook/cluster/nutanix_validator.go

```diff
@@ -125,3 +134,35 @@ func checkIfPrismCentralIPInLoadBalancerIPRange(
 
 	return nil
 }
+
+// checkIfPrismCentralAndControlPlaneIPSame checks if Prism Central and Control Plane IP are same.
+// It compares strictly IP addresses(no FQDN) and doesn't involve any network calls.
+// This is a temporary check until we have a better way to handle this by reserving IPs
+// using IPAM provider.
+func checkIfPrismCentralAndControlPlaneIPSame(
+	pcEndpoint v1alpha1.NutanixPrismCentralEndpointSpec,
+	controlPlaneEndpointSpec v1alpha1.ControlPlaneEndpointSpec,
+) error {
+	controlPlaneEndpointIP, err := netip.ParseAddr(
+		controlPlaneEndpointSpec.ControlPlaneEndpointIP(),
+	)
+	if err != nil {
+		// controlPlaneEndpointIP is strictly accepted as an IP address from user so
+		// if it is not an IP address, it is invalid.
```

Comment:
I don't think this statement is accurate.
The API does not enforce this because using a VIP is not a requirement. You could use an external LB with DNS.
https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/8c18f88f9d6accdd126106cdf0ae47f546e35c84/api/v1alpha1/common_types.go#L28-L31

Reply:
With https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/8c18f88f9d6accdd126106cdf0ae47f546e35c84/api/v1alpha1/common_types.go#L28-L31 we'll also want to check that overlap too now.
That will always be an ipv4.

Reply:
Thanks for the pointers. I was mistaken that we forgot to put API restriction to allow only ipv4.. would be good to update konvoy CLI [flag](https://github.com/mesosphere/konvoy2/blob/e512f9d5d7c9ab3cf2ec7dd944c9f7df9c8a632b/cmd/konvoy/app/flag/nutanix.go#L317) to string type.

Also, [LB range](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/8c18f88f9d6accdd126106cdf0ae47f546e35c84/api/v1alpha1/addon_types.go#L272-L278) allows ipv4 only, event though [metalb](https://github.com/metallb/metallb/blob/eeb320f15b70603ec6b93fe8ec3c290384b8a2a0/api/v1beta1/ipaddresspool_types.go#L25-L29) can accpet ipv6, CIDR, etc.

Reply:
> With https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/986 we'll also want to check that overlap too now.
That will always be an ipv4.

Did not understand what kind of overlap? I'm comparing with IP gotten from [this](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1002/files#diff-43f16565973dd5eec1b4539f863395d67ab6b52237f0ee811b439eb1512237bbR81-R91) method, did you mean to compare both [host](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/8c18f88f9d6accdd126106cdf0ae47f546e35c84/api/v1alpha1/common_types.go#L28-L31) and VIP with PC IP?

Reply:
RE the ipv4 vs ipv6 we made the decision in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/986#discussion_r1870996440

Reply:
thanks, done in b02e13ed99c30a834760ebc89de841aa7524bc58 and 7b50e2aba7b3cafee7eaa971ec52c173e73e7d7d